earthquakes/helpers.py

```diff
@@ -0,0 +1,199 @@
+        """Funtion to get data from memory.
+
+        Parameters
+        ----------
+        i_init: int
+            index of the row position where the
+            data window will be taken from
+        window_size: int, (default 150000)
+            number of rows to be taken from the data
+        """
+        logical_gate = [(i_init in inter) or (i_init + window_size in inter) for inter in self.intervals]
+        get_intervals = [x.left for x in self.intervals[logical_gate]]
+        if len(get_intervals) == 1:
+            get_files = get_intervals
+        else:
+            logical_gate_intervals = (self.index_list >= get_intervals[0]) & (self.index_list <= get_intervals[-1:])
+            get_files = self.index_list[logical_gate_intervals]
+
+        signal = pd.DataFrame([])
+        for file in get_files:
+            new_file = pd.read_pickle(os.path.join(self.save_dir, "{}.pkl".format(file)))
+            signal = pd.concat([signal, new_file], axis=0)
+        return signal.loc[i_init: i_init + window_size - 1, :]
+
+
+def create_feature_dataset_source(caller_cl, feature_computer, xcol="acoustic_data", ycol="time_to_failure",
+                                  stft=False, stft_feature_computer=None, events_id=(245829584, 307838916),
+                                  window_size=150000, step=100):
+    """Samples sequences from the data, computes features for each sequence, and stores the result
+    in a new dataframe. This function is a modification from the 'create feature' at the engineering
+    script.
+
+    Parameters
+    ----------
+    caller_sm: caller class
+    feature_computer: FeatureComputer object or similar,
+        A class that implements a method '.compute()' that takes an array and returns
+        features. It must also have an attribute 'feature_names' that shows the corresponding
+        names of the features.
+    xcol: str, optional (default: "acoustic_data"),
+        The column referring to the the signal data.
+    ycol: str, optional (default: "time_to_failure"),
+        The column referring to the target value.
+    stft: bool, optional (default: False),
+        Whether to calculate the Short Time Fourier Transform.
+    stft_feature_computer: FeatureComputer object or None,
+        The computer for stft features.
+    events_id: tuple
+        index value that define the range of values that we are
+        going to take the data from.
+    window_size: int
+        number of observations to take to get
+        information about y
+    step: int
+        number of observations to skip between ys
+
+    Returns
+    -------
+    feature_data: pd.DataFrame,
+        A new dataframe of shape (number_intervals, number of features) with the new features per sequence.
+        The index corresponds to the y position.
+    """
+    number_intervals = math.floor((events_id[1] - events_id[0])/step)
+    indices = [events_id[0] + step * i for i in range(number_intervals)]
```

Comment:
The first cycle is (0, 5506573). When I run this code with interval 5000, I get the list [0, 5000, 10000, ...., 5495000, 5500000]. So it is taking the right intervals of 5000, which is good! However, the lists end at 5500000, which means the last 6573 datapoints for an earthquake are not taken into account. This can be partially solved by adding 1 to number_intervals (since you count from 0), but then there are still 1573 datapoints not taken into account. 
There are, to my knowledge, two options:
- Or we sample the last sample of a cycle not with an interval of 5000. The list will be then [0, 5000, 10000, ...., 5500000, 5505000, 5506573]. Pro: we do not miss the last datapoints before each earthquake in the dataset.
- Or we stop at 5505000 The list will be then [0, 5000, 10000, ...., 5500000, 5505000]. Con: we never capture the last datapoints before each earthquake in the dataset.